### DISCUSSION TITLE
PDF Exports Formatting Problems

### DISCUSSION THREAD

**Discussion by jellial** [bcholmes/StarTrek2d20#13] [General]
Hi! This tool is great, but compared to the previous iteration there are a couple of small issues with the PDFs it creates.

- The text size for the Attributes and Disciplines is too big; double digit numbers get cut off on the personnel sheets.
- The Stress and Shield fields on both personnel and ship sheets now have checks instead of crosses marking unused boxes. This can make it confusing to keep track of either in the event of shields going down or excessive stress. It would be great if unused boxes could default to crosses again so it's clear.

I've attached examples of both. :)

[U.S.S._Something-starship.pdf](https://github.com/bcholmes/StarTrek2d20/files/7936237/U.S.S._Something-starship.pdf)

[Test_Johnson-sta-character.pdf](https://github.com/bcholmes/StarTrek2d20/files/7936199/Test_Johnson-sta-character.pdf)

Thanks again for your work on this invaluable tool!

**Comment by bcholmes** [comment]
Hey! Thanks for the feedback. I'll look in to both items, but just off the top of my head:

1. I don't see the text "cut off". Here's what it looks like on my (Mac) machine using (Mac's default PDF viewer) Preview: 
<img width="355" alt="Screen Shot 2022-01-25 at 1 43 24 PM" src="https://user-images.githubusercontent.com/2981347/151039231-edd01616-02e7-4d3f-b0d5-a2c0231b6470.png">
Maybe it's different for you? In general, I've been using a setting that allows most fields to auto-scale the font to fit the available space, but since the Attributes and Disciplines are pretty predictable in terms of sizing, maybe I should just force these fields to be a particular size. I'll look in to it.

2. I have that same quibble with the "checks" in the unused boxes. The code library I'm using to help me assemble the PDFs (called pdf-lib) doesn't support the crosses (I even raised an [issue](https://github.com/Hopding/pdf-lib/issues/982) on their repo about it). I've been considering removing the excess checkboxes as an interim measure -- the downside of that change would be that people couldn't really easily "change" their stress or shields as the character evolved.

**Reply by jellial** [reply]
Thanks for responding so quickly! 

1. I tested it with my friends and they couldn't replicate the issue. I'm using Adobe Reader (PC), but they are using Edge/Chrome to view. In the end, I went to Edit>Preferences>Forms and unchecked "Show text field overflow indicator". Strange that it would happen at all since as you've mentioned it's set to autoscale but unchecking that is allowing it to display the numbers properly now. 

2. Agreed; the removing the checkboxes would make it more of a hassle to increase/decrease total stress/shield capacity. Sorry to hear that the code library doesn't support alternate checks! It could get confusing as is but I think its preferable over removing excess boxes altogether.

Again thanks for making this! It's been super useful in helping my players and I with character/ship creation. ^^

**Comment by andrasbalogh93** [comment]
When I create a two page TNG style character sheet, the last word and dot in the description of "Testing a Theory" and "Former initiate" talents are missing. In the attached pdf I added the last words manually, indicated by the different fotn and size
[Lorena_Quinn-sta-character.pdf](https://github.com/bcholmes/StarTrek2d20/files/7965623/Lorena_Quinn-sta-character.pdf)

**Reply by bcholmes** [reply]
Oh, hey. That's a bug. It looks like it happens when the very last word should be on a new paragraph by itself. I'll get a fix out soon.

**Comment by kaktoforum** [comment]
@bcholmes 
Is it bug or feature?


Source "es", PDF "en"

![2024-02-14_20-01](https://github.com/bcholmes/StarTrek2d20/assets/77748199/99f6798e-b9fc-4e00-925c-bd29eebc0973)

**Reply by bcholmes** [reply]
At the moment, the PDF templates we use for character sheets are language-specific, and the clear majority of them are in English. So text like "Attributes", "Control", and so forth are static parts of the PDF, and not dynamically created. 

There's one available in German, which only shows up as an option if the user is using the web site in German.
